src/bin/tester.rs

```diff
@@ -136,6 +139,87 @@ fn ambiguity_check(handle: ModuleContentsHandle) -> Result<(), MoreThanOne> {
     Ok(())
 }
 
+#[derive(Debug)]
+struct Counters {
+    total_count: u16,
+    unambiguous_count: u16,
+    ambiguous_count: u16,
+    too_short_count: u16,
+    no_parse_count: u16,
+}
+
+impl Add for Counters {
+    type Output = Counters;
+
+    fn add(self, other: Counters) -> Counters {
+        Counters {
+            total_count: self.total_count + other.total_count,
+            unambiguous_count: self.unambiguous_count + other.unambiguous_count,
+            ambiguous_count: self.ambiguous_count + other.ambiguous_count,
+            too_short_count: self.too_short_count + other.too_short_count,
+            no_parse_count: self.no_parse_count + other.no_parse_count,
+        }
+    }
+}
+
+#[derive(Debug)]
+enum ParseResult {
+    Unambiguous,
+    Ambiguous,
+    Partial,
+    Error,
+}
+
+impl ParseResult {
+    fn compact_display(&self) -> String {
```

Comment:
Return a `&'static str` and `.to_string()` where needed instead (probably not needed to do `.to_string()` actually)?